File: task-7/Project Euler/project-euler-5.py
import math

def lcm(a, b):
    return a * b // math.gcd(a, b)

def smallest_evenly_divisible(N):
    result = 1
    for i in range(1, N + 1):
        result = lcm(result, i)
    return result

T = int(input())

for _ in range(T):
    N = int(input())
    answer = smallest_evenly_divisible(N)
    print(answer)

File: task-7/HackerRank Project Euler/project-euler-5.c
#include <stdio.h>

int gcd(int a, int b)
{
    while (b != 0)
    {
        int temp = b;
        b = a % b;
        a = temp;
    }
    return a;
}

int lcm(int a, int b)
{
    return (a * b) / gcd(a, b);
}

int smallest_evenly_divisible(int N)
{
    int result = 1;
    for (int i = 1; i <= N; i++)
    {
        result = lcm(result, i);
    }
    return result;
}

int main()
{
    int T;
    scanf("%d", &T);
    for (int t = 0; t < T; t++)
    {
        int N;
        scanf("%d", &N);
        int answer = smallest_evenly_divisible(N);
        printf("%d\n", answer);
    }
    return 0;
}